File: php/features.php
<?php
declare(strict_types=1);

// Cepik SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CepikFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CepikBaseFeature();
            case "test":
                return new CepikTestFeature();
            default:
                return new CepikBaseFeature();
        }
    }
}

File: py/features.py
# Cepik SDK feature factory

from feature.base_feature import CepikBaseFeature
from feature.test_feature import CepikTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CepikBaseFeature(),
        "test": lambda: CepikTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
